### PR TITLE
DataViews: centralize the view definition and rename `list` to `table`

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/55083

## What?

This PR:

- centralizes the view definition into a single place
- renames the `list` layout to `table` & updates its icon to `blockTable`
- renames the `side-by-side` layout to `list`

https://github.com/WordPress/gutenberg/assets/583546/e81610c3-077f-435c-9f98-8a10ee6af3ff

## Why?

The centralization helps to visualize the emerging shape of a view object, as well as helping to maintain consistency (see for example how icons drifted due to lacking a single place to update them all: [#56602](https://github.com/WordPress/gutenberg/pull/56602)).

Renaming was part of the tracking issue.

## How?

Centralizes the view definition in `constants.js`.

## Testing Instructions

- Enable the "admin views" experiment and visit "Manage all templates" or/and "Manage all pages".
- Click around everything works as before.

## Follow-ups

- Rename the react view components.

### CODE REVIEW

File: packages/edit-site/src/components/dataviews/constants.js [hunk 1]
```diff
@@ -1,5 +1,53 @@
...
 
 // Filter operators.
 export const OPERATOR_IN = 'in';
+
+// View layouts.
+export const LAYOUT_TABLE = 'table';
+export const LAYOUT_GRID = 'grid';
+export const LAYOUT_SIDE_BY_SIDE = 'side-by-side';
+
+export const VIEW_LAYOUTS = [
```
Comment (oandregal) [inline packages/edit-site/src/components/dataviews/constants.js:25]: This exercise has shown that there's quite a bit of expectations defined for a view, already. At some point in the following weeks, we could explore having official APIs for some things that were copied around and are now constants.

For example:

- `getIconForViewType( )`
- `viewTypeSupports( 'preview' )`
- etc.

**Review by youknowriad — COMMENTED** [review]
I like this PR 👍 Good start. We definitely need things like `getLayout( layout )`. I saw the icon used in the sidebar and it could use such API.

I also think we should start thinking about moving everything to a "dataviews" package. It could contain a store with layout registry for instance...

**Comment by jameskoster** [comment]
If 'side-by-side' is to become 'list', then renaming the current 'list' to 'table' seems reasonable. If that's the case we can restore the `blockTable` icon too. Sorry for the detour!

**Comment by oandregal** [comment]
> If 'side-by-side' is to become 'list', then renaming the current 'list' to 'table' seems reasonable. If that's the case we can restore the blockTable icon too. Sorry for the detour!

Thanks for the confirmation! I'll do that then.

**Comment by oandregal** [comment]
ok, this is ready then:

- centralized the view definition into a single place
- renamed the `list` layout to `table` & updates its icon to `blockTable`
- renamed the `side-by-side` layout to `list`

https://github.com/WordPress/gutenberg/assets/583546/e81610c3-077f-435c-9f98-8a10ee6af3ff

One thing I haven't done in this PR is renaming the view components. I'll do that separately so it's easier to review.

**Review by ntsekouras — APPROVED** [review]
LGTM, thank you!

**Comment by oandregal** [comment]
follow-up renaming components https://github.com/WordPress/gutenberg/pull/56709

**Pull request merged**